### ISSUE TITLE
Transcript RNG

### ISSUE THREAD

**Issue by noot** [gtank/merlin#2]
Hey, are there any plans to implement a transcript RNG like there is in the rust library?
https://github.com/dalek-cryptography/merlin/blob/master/src/transcript.rs#L280

If not, I would be interested in working on it.

**Comment by gtank** [comment]
Yes, definitely! This is a side project for me so it's slow going, but I've been trying to think of an idiomatic Go way to expose that. At the moment I'm favoring a builder type that will downscope itself to `io.Reader` when you're ready to start reading from it.

Do you have a favored approach?

**Comment by noot** [comment]
That sounds reasonable to me! Thanks so much for your work. What can I do to help get it done?

**Comment by gtank** [comment]
Hey, I'm clearly not going to get around to this as quickly as I'd hoped. I didn't mean to lick the cookie! If you would like to work on it, please go ahead. I'll be happy to look at an API or review a PR @noot

**Comment by vedhavyas** [comment]
@gtank @noot I have implemented #4 that implements `io.Reader`. Would love to get your feedback on that.
